registrations/views.py

```diff
@@ -1095,6 +1095,57 @@ def prune(self, request):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
+class FacilityCodeCheckView(APIView):
+    """ Facility Code Check
+        GET - returns clinic name and clinic code for varification
+        getting auth'd requires DB
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        if not (
+            settings.JEMBI_BASE_URL
+            and settings.JEMBI_USERNAME
+            and settings.JEMBI_PASSWORD
+        ):
+            return Response(
+                "Jembi integration is not configured properly.",
+                status.HTTP_503_SERVICE_UNAVAILABLE,
+            )
+
+        try:
+            clinic_code = request.query_params.get("clinic_code")
+            result = requests.get(
+                urljoin(settings.JEMBI_BASE_URL, "facilityCheck"),
+                headers={"Content-Type": "application/json"},
+                auth=(settings.JEMBI_USERNAME, settings.JEMBI_PASSWORD),
+                params={"criteria": "value:{}".format(clinic_code)},
+                verify=False,
+            )
+            result.raise_for_status()
+            jembi_result = result.json()
+            if len(jembi_result.get("rows")) > 0:
+                resp = {"result": {"Facility": jembi_result.get("rows")[0][2]}}
```

Comment:
We probably want to simplify this a bit, something like `{"facility": "test facility name"}`, I think getting nested fields might be a bit difficult in RapidPro.

Reply:
okay thanks